packages/remix-dev/compiler/compiler.ts

```diff
@@ -100,7 +101,8 @@ export let create = async (ctx: Context): Promise<Compiler> => {
     // js compilation (implicitly writes artifacts/js)
     let js = await tasks.js;
     if (!js.ok) throw error ?? js.error;
-    let { metafile, hmr } = js.value;
+    let { metafile, outputFiles, hmr } = js.value;
+    writes.js = JS.write(ctx.config, outputFiles);
```

Comment:
As explained in my [other comment](https://github.com/remix-run/remix/pull/7269/files#r1306751467), we now need to manage writing the output of the JS build to disk.